frontend/src/components/Login/Signup.tsx

```diff
@@ -31,15 +33,19 @@ export function Signup() {
     setUserCredentials(currentuserCredentials => ({ ...currentuserCredentials, [name]: value }));
   };
 
-  const handleSubmit = async (e: React.FormEvent<HTMLFormElement>) => {
+  const OnSignUp = async (e: React.FormEvent<HTMLFormElement>) => {
     e.preventDefault();
-    try {
-      const res = await signUp(userCredentials.email, userCredentials.password);
-      if (res) navigate('/home');
-    } catch (err) {
-      if (err instanceof FirebaseError) {
-        setError(err.message);
-      }
+    const res = await signUpApi(
+      userCredentials.name,
+      userCredentials.email,
+      userCredentials.password,
+      userCredentials.username
+    );
+    if (res.error) {
```

Comment:
If `res.error` is truthy, throw a new `Error`, with `res.error` as the `message`.

Also avoid unnecessary nesting be moving the `login` and `navigate` calls outside the `else` block. If you throw the `Error` above, you don't have to worry about these methods being called when `Promise` rejects.